laceworksdk/http_session.py

```diff
@@ -220,115 +248,141 @@ def get(self, uri, org=False):
 
         return response
 
-    def patch(self, uri, org=False, data=None, param=None):
+    def get(self, uri, params=None, **kwargs):
         """
-        :param uri: uri to send the HTTP POST request to
-        :param org: boolean representing whether the request should be performed at the Organization level
-        :param data: json object containing the data
-        :param param: python object containing the parameters
+        :param uri: uri to send the HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
         :return: response json
 
-        :raises: ApiError if unable to get a connection
+        :raises: ApiError if anything but expected response code is returned
         """
 
-        self._check_access_token()
+        # Perform a GET request
+        response = self._request("GET", uri, params=params, **kwargs)
 
-        uri = f"{self._base_url}{uri}"
+        return response
 
-        logger.info(f"PATCH request to URI: {uri}")
-        logger.debug(f"PATCH request data:\n{data}")
+    def get_pages(self, uri, params=None, **kwargs):
+        """
+        :param uri: uri to send the initial HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
-        # Perform a PATCH request
-        response = self._session.patch(uri, params=param, json=data, headers=self._get_request_headers(org_access=org))
+        :return: a generator that Gets and yields pages of data
 
-        # Validate the response
-        self._check_response_code(response, DEFAULT_SUCCESS_RESPONSE_CODES)
+        :raises: ApiError if anything but expected response code is returned
+        """
 
-        self._print_debug_response(response)
+        response = self.get(uri, params=params, **kwargs)
 
-        return response
+        while True:
+            yield response
+
+            try:
+                next_page = response.json().get('paging').get('urls').get('nextPage')
```

Comment:
nit. when chaining gets you would typically want to set a default of `{}`.  For instance: `.get('paging', {})` such that we don't attempt `.get()` on None.  However the flow control here is such that you would catch any and all of these exceptions, so you could probably switch to direct access `response.json()['paging']['urls']['nextPage']`.  Also feel free to ignore me.

Reply:
Completely agree `response.json()['paging']['urls']['nextPage']` seems a bit more pythonic given how the logic is set.  Updated in latest commits.